tests/test_utils.py

```diff
@@ -90,6 +90,24 @@ def test_safe_padded_read_padding_formats():
         assert region.shape == (8 + 2, 8 + 2)
 
 
+def test_safe_padded_read_padding_shape():
+    """Test safe_padded_read for padding shape."""
+    data = np.zeros((16, 16))
+
+    bounds = (1.5, 1, 5, 5)
+    with pytest.raises(ValueError):
+        utils.image.safe_padded_read(data, bounds, padding=(1, 1, 1))
+
+
+def test_safe_padded_read_stride_shape():
+    """Test safe_padded_read for padding size."""
+    data = np.zeros((16, 16))
+
+    bounds = (1.5, 1, 5, 5)
+    with pytest.raises(ValueError):
+        utils.image.safe_padded_read(data, bounds, stride=(1, 1, 1))
```

Comment:
Just spotted this, the stride argument is incorrect. It is a 3-tuple but it is unpacked to x and y stride value internally. This should probably error. Could also allow 3-tuple for striding channel although I cannot see that being used enough to worry about it for now.

Reply:
I am not sure if I got your question right. This was supposed to be incorrect to generate an error, right?